### ISSUE TITLE
Accurately timestamping data collected by an MCU

### ISSUE THREAD

**Issue by xjjak** [xjjak/LapCal#63]
To merge the training data, consisting of the IMU values and reference keystroke/events data, we need to accurately timestamp the data coming from a microcontroller in relation to a host (PC/laptop) that is collecting the reference data. 

**What is *accurate*?**
The bottleneck so far has been the IMU readouts, which therefore dictates frequency. Multiple reference data points/states can be assigned to a single IMU data point. If we collapse multiple reference data points into one timestamp, in the same frequency the IMU data is collected, an offset, most likely meaning latency, so negative offset, of half of the frequency should still be okay to merge the timestamps *accurately*. With the current frequency being ~100Hz, so 10ms per datapoint, a delay of 5ms would be acceptable?
The more accurate the better though...

**Current status**
- [x] Test NTP timestamp accuracy on ESP32 (Compare host timestamp with MCU's over serial transmission)
       The expected output would be the latency introduced by the serial transmission, which this can then be compared to. If this is negligible with `ms` accuracy, this opens up the possibility of timestamping data points when coming in on the host, instead of the MCU itself.
- [ ] Compare above accuracy to that achieved over BLE
       If the latency introduced by BLE is within `2-12ms`, which is in typical range (_confirmation and source needed_), it could further indicate accurate timestamping on the ESP.

**Comment by xjjak** [comment]
After customizing the firmware for a wired test setup, while writing the python scripts for data collection, I ran into the problem that my computer's clock wasn't synchronized via NTP. While fixing the problem, I read about the inaccuracy, which is tens of milliseconds, which would not be sufficient for this. 
I also looked for specific ways to synchronize microcontrollers and I was further driven to recognize:
1. NTP timestamps won't be accurate enough
2. A wired setup (before merge) is required, because BLE delays are jittery, so not even consistent
3. Because the reference data for training is generated by a microcontroller anyways, it makes more sense to connect the two microcontrollers, one reads the IMU data and the other reads the state of keyboard buttons, and the data is merged on one of them, and then sent to a host (instead of sending the data to the host, on which delays etc. are harder to control in a high level OS and merging it there)
